packages/@lwc/integration-karma/test/rendering/adjacent-text-and-comments/index.spec.js

```diff
@@ -0,0 +1,41 @@
+import { createElement } from 'lwc';
+import Preserve from 'x/preserve';
+import Ignore from 'x/ignore';
+
+it('renders adjacent text/comments without lwc:preserve-comments', async () => {
+    const elm = createElement('x-ignore', { is: Ignore });
+    document.body.appendChild(elm);
+
+    await Promise.resolve();
+    const span = elm.shadowRoot.querySelector('span');
+
+    // one text node exactly
+    expect(span.childNodes.length).toBe(1);
+    expect(span.childNodes[0].nodeType).toBe(Node.TEXT_NODE);
+    expect(span.childNodes[0].nodeValue).toBe('');
```

Comment:
This test fails without the fix, since non-static-optimized ends up with two text nodes.